traits/src/header_provider.rs

```diff
@@ -1,33 +1,45 @@
 use ckb_types::{
-    core::{BlockNumber, HeaderView},
+    core::{BlockNumber, EpochNumberWithFraction, HeaderView},
     packed::Byte32,
 };
 
 /// Trait for header storage
 pub trait HeaderProvider {
     /// Get the header of the given block hash
     fn get_header(&self, hash: &Byte32) -> Option<HeaderView>;
+}
 
-    /// Get timestamp and block_number of the corresponding block_hash, and hash of parent block
-    fn timestamp_and_parent(&self, block_hash: &Byte32) -> (u64, BlockNumber, Byte32) {
-        let header = self.get_header(block_hash).expect("parent header exist");
-        (
-            header.timestamp(),
-            header.number(),
-            header.data().raw().parent_hash(),
-        )
-    }
+/// A compact representation of header fields, used for header verification and median time calculation
+pub struct HeaderFields {
```

Comment:
Let's call it `HeaderIndexFields`?

Reply:
we have a struct named HeaderIndex already, which includes a `total_difficulty` field for comparing longest chain, there is no need for this comparison in `HeaderFields`, so I use a different name.
```
pub struct HeaderIndex {
    number: BlockNumber,
    hash: Byte32,
    total_difficulty: U256,
}
```